examples/nats-echo/main.go

```diff
@@ -172,5 +188,18 @@ func lookupGeo() string {
 	if err := json.Unmarshal(body, &g); err != nil {
 		log.Fatalf("Error unmarshalling geo: %v", err)
 	}
-	return g.Region + ", " + g.Country
+	if g.Error {
+		log.Printf("IP Geo data retrieval failed, reason %q message %q", g.Reason, g.Message)
+		return ""
+	}
+	result := g.Region + ", " + g.Country
+
+	// If the env var is used, then there might be something like ?key=... in the URL.
+	// Censor that.
+	logURL := url
+	if offset := strings.IndexByte(logURL, '?'); offset >= 0 {
```

Comment:
The std lib I am sure has something that does this already.

Reply:
From the url object, you could use a printf with `"%s://%s", u.Scheme, u.Host`

Reply:
We're not making a URL object here, and the stdlib `net/url` package got a `.Redacted() string` method fairly recently (Go 1.15, we still need to support Go 1.14, I believe?), but it only covers a password in the `user:pass@` part (userinfo within the authority), not a key in a query param.

The spec requires `"?"` before the `query` part, and as `?` is in gen-delims for the URI syntax, it has to be percent-encoded if it appears elsewhere.  So for "nuke all query params before logging", this fix is correct.

If we _also_ want to round-trip through a URL object, to redact a password there, we could, but I'm not aware of any geolocation services using user:pass in URLs for access, instead of either a query param or a header-borne token.  So I think this fix is correct and sufficient, without breaking compatibility with Go 1.14.

What I have changed is to add `"?[...]"` to the string when this happens, to make it clearer that a non-default URL which happens to be the same service provider is used.

Reply:
Why do we need the query params at all for printing out?

Reply:
To indicate that we did remove something.  When using the code with simple truncation (as in the first commit of this PR) I found myself having to double-check that I really had set the env override correctly.  Using a paid account for this geolocation API, the only difference in URL was that `?key=secretstuff` got added, which I was removing to avoid logging secrets.  So I couldn't tell that I really had deployed the env var fix.

Logging with `?[...]` in there will make it clear that an environment override really has taken effect.